[chrome-launcher] Publish type definitions instead of source TypeScript files

devrelm opened the pull request:
This _should_ fix #2896 and reduce errors caused by different ts versions, ts configs, and ts declaration file versions.

In general, this PR enables the option in chrome-launcher's tsconfig to produce declaration files and updates the .npmignore to ignore `.ts` files.

One additional change was the addition of the `@types/rimraf` package. This had to be done, as the compiler was complaining about exporting a private type.

patrickhulce commented:
I agree with publishing the `.d.ts` files instead 👍  consumers shouldn't have to deal with our typescript compilation quirks. Thanks for taking this on @devrelm!

samccone commented:
lgtm. thanks!

brendankenny commented:
looks like formatting needs to be done for this change. Can fix with `cd chrome-launcher && yarn format` (and can check locally that it worked with `yarn test-launcher-formatting`)

devrelm commented:
Yeah, I actually noticed that my change to .npmignore would ignore the `d.ts` files, so I've got a couple changes to make.

I wasn't seeing the formatting error locally, but I actually realized that it's because of a similar globbing issue: the current `*.ts` and `**/*.ts` globs used to pick up files for formatting were picking up d.ts files as well. I wasn't seeing the issue locally because I had removed them.

devrelm commented:
@brendankenny I just added a change to `./chrome-launcher/test/check-formatting.sh` to not check the formatting of *.d.ts files. Globs don't quite cut it for selecting files that match `*.ts`, but _not_ `*.d.ts`, so I switched it to use `find` instead. Let me know if you can think of a better way.

I've also updated my commit that touched `.npmignore` to (hopefully) make sure that d.ts files will be included as well.

devrelm commented:
So one final thought: I should probably also update the chrome-launcher `format` package.json script. It's similarly using `*.ts` and `**/*.ts` file expansion globs, so it'll be erroneously formatting *.d.ts files as well.

I see two options here:

1) make a similar change, calling `find`, in which case I'd probably just make a function that could be shared between the format and test scripts, so that we only have to make future updates in one place.

2) use the [`--glob`](/angular/clang-format#globbing-files) option for `clang-format`, which uses [node-glob](https://www.npmjs.com/package/node-glob) and so has more options to exclude files.

I'll go for the second option for now since it'll be quick, but I'll be happy to change it if you have a different preference.

devrelm commented:
Alright. I just did the first option, though I didn't make a shared function. Apparently importing scripts with a relative path is not a solved problem in bash, and I don't feel like going down that rabbit hole right now.

brendankenny commented:
@samccone can you take another look?

devrelm commented:
@samccone I ran `npm pack` on this locally on it appears to include the `d.ts` files and exclude the typescript files. I'm not sure if you want to sanity-check it yourself as well.
